app/node/categories/categoryCalculation.js

```diff
@@ -0,0 +1,21 @@
+const { search, exact } = require('./categoriesMap.json');
```

Comment:
The reason I had a `categoryCalculation-example` file and had the actual `categoryCalculation` file git ignored is that the contents of this file will change per person. Especially since category names are not necessarily the same for everyone.

After this change it will be a problem to have a personal mapping file because it means changing a git tracked file.

So we can either continue with some variation of having an example and real ignored file or we can implement this as a file that the app stores in the file system (like the config or part of the config). The second solution is better but will require ui to allow editing it.

WDYT?

Reply:
1. How do you expect the users will use our product? From the **source code** or from the **Release**? I'm guessing from both.
2. For **release**, we need a ready-to-use file, but now I think we can prepare the source before releasing it.
3. For running from **source**, I think we need a few preparations as possible, and it means we need a ready-to-use file in the source.
4. We already talked about creating a global default categories mapping, and per-user override mapping, I think it will solve the problems, also if the user wants other names to the same categories, we can implement that based on the global mapping.
5. I need to remember that the `require` will be bundled with `webpack`, so this file will not be available for customizing in the Release.

Maybe all these concerns require us to open an issue for decide the whole solution before the development.

Reply:
> Maybe all these concerns require us to open an issue for decide the whole solution before the development.

Agreed, I think we need some more design on the solution. I will add my thoughts to the issue and we can discuss a bit more before implementing.

To the points you mentioned:

1. Both
2. For release as well I expect the user to be able to customize his categories as he pleases which points in the direction of saving another file like the config. This can be a global with overrides but the problem with that is that it could map to categories that I as a user don't even have (in tools like ynab or other structured budgeting tools). The solution may be to have a default mapping but if the result is a category the user doesn't have just ignore it.
3. Agree that it needs to be seamless from source but that doesn't necesarily mean having a hard coded file in git. It could work just like the config - if there isn't a categories mapping, the code will create the file from the example.
4. See 2
5. Yes, thats why we should probably store this as a file in the data directory that will be available also on release and not as a file in the source code.